astropy/visualization/wcsaxes/formatter_locator.py

```diff
@@ -512,7 +512,6 @@ def __init__(
         format_unit=None,
     ):
         if unit is not None:
-            unit = unit
```

Comment:
Making only this change is just hiding bad code:
```python
if unit is None:
    if spacing is not None:
        unit = spacing.unit
    elif values is not None:
        unit = values.unit
format_unit = format_unit or unit
```
If it is the case the `unit`, `spacing` and `values` can't all be `None` then this could be simplified even more.

Reply:
Done.

Reply:
Yeah, a nice simplification might be
``unit = unit if unit is not None else getattr(spacing, "unit", getattr(values, "unit", None))``

Reply:
@eerovaher, let's consider this in a future PR. I think this is the last outstanding PR comment.

Reply:
We have already simplified the code here, so this should be good to merge.

Reply:
Yes, and in this case I don't think I'd consider the one-line alternative easier to parse, so that should certainly be discussed separately!